### PR TITLE
JavadocMetadataScrapperTest and InlineConfigParser fixes

### PR DESCRIPTION
**Pull request by rnveach**
### reworks JavadocMetadataScrapperTest to make more sense

Moved all tests related to `JavadocMetadataScrapper` into `JavadocMetadataScrapperTest` and use BDD and actually verify outputs from the check. Hopefully, at some point we can use pitest and such on this class to verify it's code.

### changes InlineConfigParser to respect fully qualified name 

This is a simple change to so that when it gets the fully qualified name, it takes as the fully qualified name and doesn't make any other changes. This is all because `JavadocMetadataScrapper` doesn't end with `Check` so BDD can't recognize it. No idea why the class was named this, but BDD should still be changed as I expect the FQN to be the FQN.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java [hunk 1]
```diff
@@ -213,10 +227,7 @@ else if (ast.getType() == JavadocTokenTypes.SINCE_LITERAL) {
     public void finishJavadocTree(DetailNode rootAst) {
         moduleDetails.setDescription(getDescriptionText());
         if (isTopLevelClassJavadoc()) {
-            if (getFileContents().getFileName().contains("test")) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java:216]: This is the main reason the property is needed to drop having this code that checks the file name which will cause issues with checks with that word in them. New property always defaults to `write`.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java [hunk 2]
```diff
@@ -603,13 +617,18 @@ private static String getPackageName(String filePath) {
         return Collections.unmodifiableMap(MODULE_DETAILS_STORE);
     }
 
+    /** Reset the module detail store of any previous information. */
+    public static void resetModuleDetailsStore() {
+        MODULE_DETAILS_STORE.clear();
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java:622]: Tests are writing to a global static field. Nothing ensure this field is cleared, so this reset method is there to ensure the field is cleared.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java [hunk 3]
```diff
@@ -234,7 +248,7 @@ public void finishJavadocTree(DetailNode rootAst) {
      *
      * @param ast javadoc ast
      */
-    public void scrapeContent(DetailNode ast) {
+    private void scrapeContent(DetailNode ast) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java:251]: No reason this method should be `public`.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java [hunk 4]
```diff
@@ -603,13 +617,18 @@ private static String getPackageName(String filePath) {
...
+    }
+
     /**
      * Check if the current javadoc block comment AST corresponds to the top-level class as we
      * only want to scrape top-level class javadoc.
      *
      * @return true if the current AST corresponds to top level class
      */
-    public boolean isTopLevelClassJavadoc() {
+    private boolean isTopLevelClassJavadoc() {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java:631]: Same.

File: src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraperTest.java [hunk 5]
```diff
@@ -19,34 +19,225 @@
...
+                    CommonUtil.EMPTY_STRING_ARRAY);
+            fail("Exception expected");
+        });
+        assertThat(exc.getCause()).isInstanceOf(MetadataGenerationException.class);
+        assertThat(exc.getCause().getMessage())
+                .isEqualTo("Type for property 'misplacedType' is missing");
     }
 
     @Test
-    public void testPropertyNameWithNoCodeTag() {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraperTest.java:45]: Removed test basically got encompassed into the new tests.

File: src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraperTest.java [hunk 6]
```diff
@@ -19,34 +19,225 @@
...
+        return builder.toString();
+    }
+
+    private static void append(StringBuilder builder, String title, Object object) {
+        builder.append(title).append(object).append('\n');
+    }
 
-        assertNull(testCheckMeta.getProperties().get(0).getName(),
-                "Check with no property name {@code } tag should have null property name");
+    private static String split(String string, int size) {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraperTest.java:225]: Split code is to help ensure line does not exceed the 100 character limit.

File: src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java [hunk 7]
```diff
@@ -76,23 +78,18 @@ public static void generate(String... args) throws IOException, CheckstyleExcept
     private static void dumpMetadata(Checker checker, String path) throws CheckstyleException,
             IOException {
         final List<File> validFiles = new ArrayList<>();
-        if (path.endsWith(".java")) {
-            validFiles.add(new File(path));
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java:80]: Removed testing single files which I assume was only used by UTs. Most of our needs scan folders.

**Review by strkkk — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Review by romani — APPROVED** [review]
ok to merge

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**